Form/ChoiceList/ModelChoiceList.php

```diff
@@ -105,7 +105,7 @@ public function __construct($class, $labelPath = null, $choices = null, $queryOb
             $this->identifier = $this->query->getTableMap()->getPrimaryKeys();
         }
 
-        $this->loaded = is_array($choices) || $choices instanceof \Traversable;
+        $this->loaded = (is_array($choices) && array() !== $choices) || $choices instanceof \Traversable;
```

Comment:
this is wrong. If the user passes an empty array explicitly, it must be used as the loaded choices